pkg/actors/actor.go

```diff
@@ -7,12 +7,60 @@ package actors
 
 import (
 	"sync"
+	"sync/atomic"
 	"time"
+
+	diag "github.com/dapr/dapr/pkg/diagnostics"
 )
 
 type actor struct {
-	lock         *sync.RWMutex
-	lastUsedTime time.Time
-	busy         bool
-	busyCh       chan (bool)
+	actorType string
+	actorID   string
+
+	concurrencyLock *sync.RWMutex
+	lastUsedTime    time.Time
+	busy            bool
+	busyCh          chan (bool)
+
+	pendingLockCount int32
+}
+
+func newActor(actorType, actorID string) *actor {
+	return &actor{
+		actorType:       actorType,
+		actorID:         actorID,
+		concurrencyLock: &sync.RWMutex{},
+		busy:            false,
```

Comment:
It looks like this should be true?  After this line which initializes it

val, _ := a.actorsTable.LoadOrStore(key, newActor(actorTypeID.GetActorType(), actorTypeID.GetActorId()))

the actor method is called.  I don't see `busy` being set before that.

Reply:
act.lock() is called in callLocalActor() and lock() will set busy to `true`.. please check lock() method.

Reply:
Oh it's set to true inside lock now.

Reply:
technically, why should actor be busy before user request gets lock?

Reply:
ok.